fluent-syntax/src/parser.js

```diff
@@ -415,96 +415,106 @@ export default class FluentParser {
   }
 
   getPattern(ps) {
-    let buffer = '';
     const elements = [];
-    let firstLine = true;
+    ps.skipInlineWS();
 
-    if (this.withSpans) {
-      var spanStart = ps.getIndex();
+    // Special-case: trim leading whitespace and newlines.
+    if (ps.isPeekNextLinePattern()) {
+      ps.skipWS();
     }
 
     let ch;
     while ((ch = ps.current())) {
-      if (ch === '\n') {
-        if (firstLine && buffer.length !== 0) {
-          break;
-        }
 
+      // The end condition for getPattern's while loop is a newline
+      // which is not followed by a valid pattern continuation.
+      if (ch === '\n' && !ps.isPeekNextLinePattern()) {
+        break;
+      }
+
+      if (ch === '{') {
```

Comment:
sorry, I should have state my opinion. I don't believe that it's necessary. I think it only looks weird because of the semantic meaning you give to `}` when seen in this context.
I'd argue that:

```
foo = Bar = Baz
```

looks equally weird. :)